homeassistant/components/screenlogic/sensor.py

```diff
@@ -85,190 +145,97 @@
     STATE_TYPE.TOTAL_INCREASING: SensorStateClass.TOTAL_INCREASING,
 }
 
-SL_UNIT_TO_HA_UNIT = {
-    UNIT.CELSIUS: UnitOfTemperature.CELSIUS,
-    UNIT.FAHRENHEIT: UnitOfTemperature.FAHRENHEIT,
-    UNIT.MILLIVOLT: UnitOfElectricPotential.MILLIVOLT,
-    UNIT.WATT: UnitOfPower.WATT,
-    UNIT.HOUR: UnitOfTime.HOURS,
-    UNIT.SECOND: UnitOfTime.SECONDS,
-    UNIT.REVOLUTIONS_PER_MINUTE: REVOLUTIONS_PER_MINUTE,
-    UNIT.PARTS_PER_MILLION: CONCENTRATION_PARTS_PER_MILLION,
-    UNIT.PERCENT: PERCENTAGE,
-}
-
 
 async def async_setup_entry(
     hass: HomeAssistant,
     config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up entry."""
-    entities: list[ScreenLogicSensorEntity] = []
+    entities: list[ScreenLogicSensor] = []
     coordinator: ScreenlogicDataUpdateCoordinator = hass.data[DOMAIN][
         config_entry.entry_id
     ]
-    equipment_flags = coordinator.gateway_data[SL_DATA.KEY_CONFIG]["equipment_flags"]
-
-    # Generic push sensors
-    for sensor_name in coordinator.gateway_data[SL_DATA.KEY_SENSORS]:
-        if sensor_name in SUPPORTED_BASIC_SENSORS:
-            entities.append(
-                ScreenLogicStatusSensor(coordinator, sensor_name, CODE.STATUS_CHANGED)
+    gateway = coordinator.gateway
+
+    for base_data in process_supported_values(gateway, SUPPORTED_DATA):
+        base_kwargs = {
+            "data_path": base_data.data_path,
+            "key": base_data.entity_key,
+            "device_class": SL_DEVICE_TYPE_TO_HA_DEVICE_CLASS.get(
+                base_data.value_data.get(ATTR.DEVICE_TYPE)
+            ),
+            "entity_category": base_data.value_parameters.get(
+                EntityParameter.ENTITY_CATEGORY, EntityCategory.DIAGNOSTIC
+            ),
+            "entity_registry_enabled_default": base_data.enabled,
```

Comment:
It looks like this block appears a few times. It could be made a bit more DRY

Reply:
I pushed all shared key setup into `process_supported_values()` [a3710a0](https://github.com/home-assistant/core/pull/92475/commits/a3710a0a6c6133c28ef224c7053a6c9ef1f298bf)